packages/@n8n/nodes-langchain/nodes/mcp/McpClientTool/utils.ts

```
@@ -155,26 +156,46 @@ export async function connectMcpClient({
 			return createResultError({ type: 'invalid_url', error: endpoint.error });
 		}
 
-		const transport = new SSEClientTransport(endpoint.result, {
-			eventSourceInit: {
-				fetch: async (url, init) =>
-					await fetch(url, {
-						...init,
-						headers: {
-							...headers,
-							Accept: 'text/event-stream',
-						},
-					}),
-			},
-			requestInit: { headers },
-		});
-
 		const client = new Client(
 			{ name, version: version.toString() },
 			{ capabilities: { tools: {} } },
 		);
 
-		await client.connect(transport);
+		try {
+			const transport = new StreamableHTTPClientTransport(endpoint.result, {
+				eventSourceInit: {
+					fetch: async (url, init) =>
+						await fetch(url, {
+							...init,
+							headers: {
+								...headers,
+								Accept: 'text/event-stream',
+							},
+						}),
+				},
+				requestInit: { headers },
+			});
+			await client.connect(transport);
+			console.log('Connected using Streamable HTTP transport');
+		} catch (error) {
+			// If that fails with a 4xx error, try the older SSE transport
```

Falling back like this seems suboptimal and introduces unpredictability. It might be better to let the user explicitly configure the transport mode (e.g., via an option ), rather than attempting one and silently falling back on failure.

---

Sure